src/App.jsx

```diff
@@ -3,41 +3,92 @@ import { RouterProvider, createBrowserRouter } from "react-router-dom";
 import api from "../api";
 import Homepage from "./pages/home-page";
 import ErrorPage from "./pages/error-page";
+import ArticlePage from "./pages/article-page";
+import Layout from "./components/layout";
 import { CircularProgress } from "@mui/material";
 
-async function fetchArticles() {
-  const { data } = await api.getArticles;
-  return data;
-}
+function App() {
+  const [articles, setArticles] = useState([]);
+  const [votes, setVotes] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchArticles = async () => {
+    try {
+      const { data } = await api.getArticles();
+      setArticles(data);
+      const initialVotes = data.reduce((acc, article) => {
+        acc[article.article_id] = article.votes;
+        return acc;
+      }, {});
+      setVotes(initialVotes);
+    } catch (error) {
+      console.error("Failed to fetch articles", error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchArticles();
+  }, [fetchArticles]);
+
+  const handleVote = async (articleId, increment) => {
+    try {
+      await api.patchVotes(articleId, increment);
+      setVotes((prevVotes) => ({
+        ...prevVotes,
+        [articleId]: prevVotes[articleId] + increment,
```

Comment:
For voting, it would be better if it's optimistic rendering instead of waiting the result to come back from the api in case some slow internet which will slowing showing the up-to-date votes